engine/src/main/java/org/apache/hop/workflow/action/ActionBase.java

```diff
@@ -738,12 +738,10 @@ public void setPluginId( String configId ) {
    * appropriate class name
    *
    * @return full class name of the dialog
-   * @deprecated As of release 8.1, use annotated-based dialog instead {@see org.apache.hop.core.annotations.PluginDialog}
    */
-  @Deprecated
   public String getDialogClassName() {
     String className = getClass().getCanonicalName();
-    className = className.replaceFirst( "\\.hop\\.", ".hop.ui." );
+    //className = className.replaceFirst( "\\.hop\\.", ".hop.ui." );
```

Comment:
We have 2 options:
1) rename package in gui without '.ui.' 
2) fallback but is not easy with classloader: try without replace string, if class not found try with replace

Reply:
I really hate the idea to change the package names in hop.ui
This works, it is not the cleanest code but it works... we should also take a look why this piece of code is still being used by the plugins. if you look at getActionDialog() it is written expecting the plugins to not use this method
` public String getDialogClassName() {
    String className = getClass().getCanonicalName();
    className += "Dialog";

    try {
      Class clazz = Class.forName(className.replaceFirst( "\\.hop\\.", ".hop.ui." ));
      className = clazz.getName();
    }catch (ClassNotFoundException e){
      //do nothing and return plugin classname
    }

    return className;
  }`

Reply:
An elegant solution is to overwrite method getDialogClassName for exception (when IAction dialog is not in the same package).
We should change HopGuiWorkflowActionDelegate.getActionDialog() to only use getDialogClassName.

Reply:
we should definitely fix this further upstream, I was thinking of using this as a temp fix to not block us. Cleanest would be to add the dialog to the plugin.classmap I think this will have the least impact on the code. Feel free to disagree